NyaChan.py

```diff
@@ -1,80 +1,108 @@
-        await ctx.author.send("{}, only my Owner can ask me to do that, nya!```{}```".format(ctx.message.author.mention,
-                                                                                             ctx.message.content))
-    elif isinstance(error, discord.ext.commands.errors.UserInputError):
-        await ctx.author.send(
-            "{}, Input error```py\n{}: {}\n```".format(ctx.message.author.mention, type(error).__name__, str(error)))
-    elif isinstance(error, discord.ext.commands.errors.NoPrivateMessage):
-        await ctx.author.send(
-            "{}, this command cannot be send in a PM!```{}```".format(ctx.message.author.mention, ctx.message.content))
-    elif isinstance(error, discord.ext.commands.errors.CheckFailure):
-        await ctx.author.send(
-            "You don\'t have the permission to use this command, {}```{}```".format(ctx.message.author.mention,
-                                                                                    ctx.message.content))
-    else:
-        await ctx.author.send(
-            "{}, error```py\n{}: {}\n```".format(ctx.message.author.mention, type(error).__name__, str(error)))
-    has_mention = False
-    for arg in ctx.args:
-        if '@' in str(arg):
-            has_mention = True
-            break
-    if has_mention is True:
-        return False
-    await ctx.message.delete()
+async def on_command_completion(ctx):
+    # Deletes message if no mentions
+    if not ctx.message.mentions:
+        try:
+            await ctx.message.delete()
+        except discord.Forbidden:
+            # Can't execute on DM channel
+            pass
 
 
 @bot.event
-async def on_command_completion(ctx):
-    has_mention = False
-    for arg in ctx.args:
-        if '@' in str(arg):
-            has_mention = True
-            break
-    if has_mention is True:
-        return False
-    await ctx.message.delete()
+async def on_error(_, msg):
+    if not os.getenv("DEBUG"):
+        return
+
+    cls, exc, tb = sys.exc_info()
+
+    if cls == ThrowawayException:
+        return
+
+    await msg.add_reaction("🚫")
+    err_str = str(cls) + ": " + str(exc.args[1]) + "\n\nTraceback:\n"
+    ttb = []
+
+    sep = "\\" if os.name == "nt" else "/"
+    for ln in traceback.format_tb(tb, 50):
+        ln_m = re.match("  File \"(.*)\", line (\d+), in ([^\s]+)\n(.*)", ln, re.DOTALL)
+
+        p = ln_m.group(1).split(sep)
+        lnn = ln_m.group(2)
+
+        tp = []
+        for i in p:
+            if not tp:
+                if i != "site-packages":
+                    continue
+                i = "<pip_pkg>"
+            tp.append(i)
+
+        ttb.append("  {} : {} ({})\n{} {}".format(
+            sep.join(tp or p), lnn, ln_m.group(3), " " if tp else ">", ln_m.group(4)))
+
+    err_str += "".join(ttb)
+    print(err_str)
 
 
 class MainDriver:
     def __init__(self, bot):
         self.bot = bot
         for cog in self.bot.config.bot.cogs:
```

Comment:
can we add a state here where if no cogs are specified we load all the cogs? or fail hard on empty list depending on your preference.  But [] is a bad state

Reply:
that shouldn't be hard. ill add that now

Reply:
+1